dropwizard-jersey/src/main/java/io/dropwizard/jersey/params/AbstractParam.java

```diff
@@ -117,7 +117,7 @@ public boolean equals(Object obj) {
         if (this == obj) {
             return true;
         }
-        if ((obj == null) || (getClass() != obj.getClass())) {
+        if (!(obj instanceof AbstractParam)) {
```

Comment:
FYI, these methods are now broken (they violate the equals contract of `a.equals(b) == b.equals(a)` when a and b both share the same `AbstractParam` interface and either of them overrides `equals`).

The only way to use `instanceof` in an equals check without violating the equals contract is to make your equals method `final`. If any subclass overrides the equals method (anything more than `return super.equals(object)`) you have broken symmetry (unless you have some ridiculously painful bits of code in the super implementation that guarantees that no subtype of your overridden class can ever be equal to any subtype with a different heritage -- and good luck maintaining that without constantly introducing equals symmetry bugs).